singularity-eos/eos/eos_spiner.hpp

```diff
@@ -14,1997 +14,12 @@
-    const Real rho, const Real T, const DataBox &db, Indexer_t &&lambda) const {
-  const Real lRho = toLog_(rho, lRhoOffset_);
-  const Real lT = toLog_(T, lTOffset_);
-  if (!variadic_utils::is_nullptr(lambda)) {
-    IndexerUtils::Get<IndexableTypes::LogDensity>(lambda, Lambda::lRho) = lRho;
-  }
-  return db.interpToReal(lRho, lT);
-}
-
-template <typename Indexer_t>
-PORTABLE_INLINE_FUNCTION Real SpinerEOSDependsRhoSie::interpRhoSie_(
-    const Real rho, const Real sie, const DataBox &db, Indexer_t &&lambda) const {
-  const Real lRho = toLog_(rho, lRhoOffset_);
-  const Real lE = toLog_(sie, lEOffset_);
-  if (!variadic_utils::is_nullptr(lambda)) {
-    IndexerUtils::Get<IndexableTypes::LogDensity>(lambda, Lambda::lRho) = lRho;
-  }
-  return db.interpToReal(lRho, lE);
-}
-
-template <typename Indexer_t>
-PORTABLE_INLINE_FUNCTION Real SpinerEOSDependsRhoSie::lRhoFromPlT_(
-    const Real P, const Real lT, Indexer_t &&lambda) const {
-  const RootFinding1D::RootCounts *pcounts =
-      (memoryStatus_ == DataStatus::OnDevice) ? nullptr : &counts;
-  RootFinding1D::Status status = RootFinding1D::Status::SUCCESS;
-  Real lRho;
-  Real dPdRhoMax = dPdRhoMax_.interpToReal(lT);
-  Real PMax = PlRhoMax_.interpToReal(lT);
-  if (dPdRhoMax > 0 && P > PMax) {
-    Real rho = (P - PMax) / dPdRhoMax + rhoMax_;
-    lRho = toLog_(rho, lRhoOffset_);
-    if (pcounts != nullptr) {
-      pcounts->increment(0);
-    }
-  } else {
-    Real lRhoGuess = reproducible_ ? lRhoMin_ : 0.5 * (lRhoMin_ + lRhoMax_);
-    if (!variadic_utils::is_nullptr(lambda)) {
-      Real lRho_cache =
-          IndexerUtils::Get<IndexableTypes::LogDensity>(lambda, Lambda::lRho);
-      if ((lRhoMin_ <= lRho_cache) && (lRho_cache <= lRhoMax_)) {
-        lRhoGuess = lRho_cache;
-      }
-    }
-    const callable_interp::l_interp PFunc(dependsRhoT_.P, lT);
-    status = ROOT_FINDER(PFunc, P, lRhoGuess, lRhoMin_, lRhoMax_, robust::EPS(),
-                         robust::EPS(), lRho, pcounts);
-    if (status != RootFinding1D::Status::SUCCESS) {
-#if SPINER_EOS_VERBOSE
-      std::stringstream errorMessage;
-      errorMessage << "inverting P table for logRho failed\n"
-                   << "matid     = " << matid_ << "\n"
-                   << "lT        = " << lT << "\n"
-                   << "P         = " << P << "\n"
-                   << "lRhoGuess = " << lRhoGuess << std::endl;
-      EOS_ERROR(errorMessage.str().c_str());
-#endif // SPINER_EOS_VERBOSE
-      lRho = reproducible_ ? lRhoMin_ : lRhoGuess;
-    }
-  }
-  if (!variadic_utils::is_nullptr(lambda)) {
-    IndexerUtils::Get<IndexableTypes::LogDensity>(lambda, Lambda::lRho) = lRho;
-    if constexpr (variadic_utils::is_indexable_v<Indexer_t, IndexableTypes::RootStatus>) {
-      lambda[IndexableTypes::RootStatus()] = static_cast<Real>(status);
-    }
-  }
-  return lRho;
-}
-
-} // namespace singularity
+#ifdef SINGULARITY_USE_SPINER_WITH_HDF5
 
 #endif // SINGULARITY_USE_SPINER_WITH_HDF5
```

Comment:
Both new spiner header files should be wrapped in these guards, but they don't add anything in this file. They're there to prevent the code from being compiled when hdf5 isn't available.